codegen/module.go

```diff
@@ -1190,13 +1190,16 @@ func (system *ModuleSystem) IncrementalBuild(
 	for _, className := range system.classOrder {
 		var wg sync.WaitGroup
 		wg.Add(len(resolvedModules[className]))
-		ch := make(chan *ModuleInstance, len(resolvedModules[className]))
+		ch := make(chan *populateSpecRes, len(resolvedModules[className]))
 		for _, instance := range resolvedModules[className] {
 			go func(instance *ModuleInstance) {
 				defer wg.Done()
 				if err := system.populateSpec(instance); err != nil {
-					// HACK: to get get of bad modules, which should not be even be loaded in dag at first place
-					ch <- instance
+					if _, ok := err.(*ErrorSkipCodeGen); ok {
+						// HACK: to get get of bad modules, which should not be even be loaded in dag at first place
```

Comment:
probably should update this comment to explain clearly as what is considered a bad module.

Reply:
error type name which is explanatory with a godoc comment on that, now I feel is sufficient. avoiding comment explanation here again to abstract those reasons to modules that may do it in different conditions and thus suited there.